pkg/providers/tinkerbell/assert.go

```diff
@@ -216,50 +217,146 @@ func MinimumHardwareAvailableAssertionForCreate(catalogue *hardware.Catalogue) C
 	}
 }
 
+// WorkerNodeHardware holds machine deployment name, replica count and hardware selector for a Tinkerbell worker node.
+type WorkerNodeHardware struct {
+	MachineDeploymentName string
+	Replicas              int
+	HardwareSelector      v1alpha1.HardwareSelector
+}
+
+// ValidatableCluster allows assertions to pull worker node and control plane information.
+type ValidatableCluster interface {
+	WorkerNodeHardwareGroups() []WorkerNodeHardware
+	ControlPlaneReplicaCount() int
+	ControlPlaneHardwareSelector() v1alpha1.HardwareSelector
```

Comment:
I would document these, it's almost more important than the implementations themselves, since this is what defines the contract.

Reply:
Added documentation for these functions